plugins/vendor/zoom/ZoomDropFolderPlugin/batch/KZoomDropFolderEngine.php

```diff
@@ -0,0 +1,423 @@
+		}
+		else
+		{
+			KalturaLog::info('No physical files found for drop folder id ['.$this->dropFolder->id.']');
+			$meetings = array();
+		}
+		
+		KalturaLog::info("physical files: ");
+		foreach ($meetings as $meeting)
+		{
+			KalturaLog::info('Meeting UUID: '. $meeting[self::UUID]);
+		}
+		return $meetings;
+	}
+
+	protected function handleMeetingFiles($meetingFiles, &$dropFolderFilesMap)
+	{
+		foreach ($meetingFiles as $meetingFile)
+		{
+			KalturaLog::debug("meeting file is: " . print_r($meetingFile, true));
+			$recordingFilesOrdered = self::orderRecordingFiles($meetingFile[self::RECORDING_FILES]);
+			KalturaLog::debug("recording files are: " . print_r($recordingFilesOrdered, true));
+			foreach ($recordingFilesOrdered as $recordingFilesPerTimeSlot)
+			{
+				$firstDFFileOnTimeSlot = true;
+				$isParentEntry = false;
+				foreach ($recordingFilesPerTimeSlot as $recordingFile)
+				{
+					$recordingFileName = $meetingFile[self::UUID] . '_' . $recordingFile[self::ID] . ZoomHelper::SUFFIX_ZOOM;
+					if (!array_key_exists($recordingFileName, $dropFolderFilesMap))
+					{
+						if (ZoomHelper::shouldHandleFileType($recordingFile[self::RECORDING_FILE_TYPE]))
+						{
+							if ($firstDFFileOnTimeSlot)
+							{
+								$firstDFFileOnTimeSlot = false;
+								$entry = $this->getEntryByReferenceId(zoomProcessor::ZOOM_PREFIX . $meetingFile[self::UUID]);
+								if ($entry)
+								{
+									$isParentEntry = true;
+								}
+								else
+								{
+									$entry = $this->createEntry($meetingFile[self::UUID]);
+								}
+							}
+							if (!$isParentEntry && $recordingFile[self::RECORDING_FILE_TYPE] == 'MP4')
+							{
+								$isParentEntry = true;
+								$this->addDropFolderFile($meetingFile, $recordingFile, $entry->id, true);
+							}
+							else
+							{
+								$this->addDropFolderFile($meetingFile, $recordingFile, $entry->id);
+							}
+							self::$lastHandledMeetingTime = self::convertTimeToUnix($meetingFile[self::START_TIME]);
+						}
+					}
+					else
+					{
+						$dropFolderFile = $dropFolderFilesMap[$recordingFileName];
+						unset($dropFolderFilesMap[$recordingFileName]);
+						$this->handleExistingDropFolderFile($dropFolderFile);
+					}
+				}
+			}
+		}
+	}
+	
+	protected function getEntryByReferenceId($referenceId)
+	{
+		try
+		{
+			$entryFilter = new KalturaBaseEntryFilter();
+			$entryFilter->referenceIdEqual = $referenceId;
+			
+			$entryPager = new KalturaFilterPager();
+			$entryPager->pageSize = 1;
+			$entryPager->pageIndex = 1;
+			KBatchBase::impersonate($this->dropFolder->partnerId);
+			$entryList = KBatchBase::$kClient->baseEntry->listAction($entryFilter, $entryPager);
+			KBatchBase::unimpersonate();
+			if (is_array($entryList->objects) && isset($entryList->objects[0]) )
+			{
+				return $entryList->objects[0];
+			}
+			
+			return null;
+		}
+		catch (Exception $e)
+		{
+			KalturaLog::err('Failed to get entry by reference id: '. $referenceId . $e->getMessage() );

```

Comment:
if we get exception here, it will not be due to now entry, but due to SW error, then please remove the catch, we prefer it to break and see that the job was closed with error, rather then hid the error.